modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+    return dates
+        .stream()
+        .reduce(Pair.of(LocalDate.MAX, LocalDate.MIN), (p, ld) -> {
+          LocalDate earliest = p.getFirst();
+          LocalDate latest = p.getSecond();
+          return Pair.of(ld.isBefore(earliest) ? ld : earliest, ld.isAfter(latest) ? ld : latest);
+        }, (p1, p2) -> p1);
+  }
+
+  public static DenseLocalDateDoubleTimeSeries of(LocalDate date, double value) {
+    return of(date, value, SKIP_WEEKENDS);
+  }
+
+  public static DenseLocalDateDoubleTimeSeries of(
+      LocalDate date,
+      double value,
+      DenseTimeSeriesCalculation dateCalculation) {
+    return new DenseLocalDateDoubleTimeSeries(date, new double[]{value}, dateCalculation);
+  }
+
+  /**
+   * Create a new time series with the specified additional points.
+   *
+   * @param additionalPoints  the points to add or update in the time series.
+   * @return a new time series containing updated points
+   */
+  // Should this not be here with the expectation that we always go through the builder?
+  public DenseLocalDateDoubleTimeSeries update(List<LocalDateDoublePoint> additionalPoints) {
+
+    // Assume points are at the end for now
+    LocalDate latestDate = additionalPoints.stream()
+        .reduce(LocalDate.MIN, (ld, pt) -> ld.isAfter(pt.getDate()) ? ld : pt.getDate(), (ld1, ld2) -> ld1);
+
+    int size = dateCalculation.calculatePosition(startDate, latestDate) + 1;
+
+    double[] values = Arrays.copyOf(points, size);
+    if (points.length + additionalPoints.size() != size) {
+      // We have gaps - prefill with NaNs
+      Arrays.fill(values, points.length, size, Double.NaN);
+    }
+
+    additionalPoints.stream()
+        .forEach(p -> values[dateCalculation.calculatePosition(startDate, p.getDate())] = p.getValue());
+
+    return new DenseLocalDateDoubleTimeSeries(startDate, values);
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return size() == 0;
+  }
+
+  @Override
+  public int size() {
+    return (int) validIndices().count();
+  }
+
+  @Override
+  public DenseLocalDateDoubleTimeSeries subSeries(LocalDate startInclusive, LocalDate endExclusive) {
+
+    ArgChecker.notNull(startInclusive, "startInclusive");
+    ArgChecker.notNull(endExclusive, "endExclusive");
+    if (endExclusive.isBefore(startInclusive)) {
+      throw new IllegalArgumentException(
+          "Invalid sub series, end before start: " + startInclusive + " to " + endExclusive);
+    }
+
+    // special case when this is empty or when the dates are the same
+    // or the series don't intersect
+    if (isEmpty() || startInclusive.equals(endExclusive) ||
+        !startDate.isBefore(endExclusive) ||
+        startInclusive.isAfter(getLatestDate())) {
+      return EMPTY_SERIES;
+    }
+
+    int startIndex = dateCalculation.calculatePosition(startDate, startInclusive);
+    int endIndex = dateCalculation.calculatePosition(startDate, endExclusive);
+
+    return new DenseLocalDateDoubleTimeSeries(
+        startInclusive,
```

Comment:
I suspect this may break with weekends, where the indices will have been weekend adjusted, but this date isn't.